test/differentiation.jl

```diff
@@ -0,0 +1,56 @@
+using DelayDiffEq, Calculus, ForwardDiff
+using LinearAlgebra, Test
+
+# Hutchinson's equation
+function f(du, u, h, p, t)
+  du[1] = p[1] * u[1] * (1 - h(p, t - p[2])[1] / p[3])
+  nothing
+end
+
+h(p, t) = [p[4]]
+
+@testset "Gradient" begin
+  function test(p)
+    prob = DDEProblem(f, [p[5]], h, eltype(p).((0.0, 20.0)), copy(p);
+                      constant_lags = [p[2]])
+    sol = solve(prob, MethodOfSteps(Tsit5()))
```

Comment:
you have to set the tolerance really low to test the difference here, since finite difference is really sensitive.

Reply:
Ah OK. However, I guess it does not matter at the moment - the values are off by multiple orders of magnitude, and even the signs do not match.

Reply:
yeah that's bad haha